src/supermarket/app/controllers/api/v1/quality_metrics_controller.rb

```diff
@@ -0,0 +1,171 @@
+  #
+  # This endpoint expects +cookbook_name+, +cookbook_version+,
+  # +foodcritic_failure+, +foodcritic_feedback+, and +fieri_key+.
+  #
+
+  def foodcritic_evaluation
+    require_evaluation_params
+
+    create_metric(
+      @cookbook_version,
+      QualityMetric.foodcritic_metric,
+      params[:foodcritic_failure],
+      params[:foodcritic_feedback]
+    )
+
+    head 200
+  end
+
+  #
+  # POST /api/v1/quality_metrics/collaborators_evaluation
+  #
+  # Take the collaborators evaluation results from Fieri and store them on the
+  # applicable +CookbookVersion+.
+  #
+  # If the +CookbookVersion+ does not exist, render a 404 not_found.
+  #
+  # If the request is unauthorized, render unauthorized.
+  #
+  # This endpoint expects +cookbook_name+, +cookbook_version+,
+  # +collaborators_failure+, +collaborators_feedback+, and +fieri_key+.
+  #
+  def collaborators_evaluation
+    require_collaborator_params
+
+    create_metric(
+      @cookbook_version,
+      QualityMetric.collaborator_num_metric,
+      params[:collaborator_failure],
+      params[:collaborator_feedback]
+    )
+
+    head 200
+  end
+
+  #
+  # POST /api/v1/cookbook-versions/publish_evaluation
+  #
+  # Take the publish evaluation results from Fieri and store them as a
+  # metric result
+  #
+  # If the +CookbookVersion+ does not exist, render a 404 not_found.
+  #
+  # If the request is unauthorized, render unauthorized.
+  #
+  # This endpoint expects +cookbook_name+, +cookbook_version+,
+  # +publish_failure+, +publish_feedback+, and +fieri_key+.
+  #
+  def publish_evaluation
+    create_metric(
+      @cookbook_version,
+      QualityMetric.publish_metric,
+      params[:publish_failure],
+      params[:publish_feedback]
+    )
+
+    head 200
+  end
+
+  #
+  # POST /api/v1/cookbook-versions/license_evaluation
+  #
+  # Take the license evaluation results from Fieri and store them as a
+  # metric result
+  #
+  # If the +CookbookVersion+ does not exist, render a 404 not_found.
+  #
+  # If the request is unauthorized, render unauthorized.
+  #
+  # This endpoint expects +cookbook_name+, +cookbook_version+,
+  # +license_failure+, +license_feedback+, and +fieri_key+.
+  #
+  def license_evaluation
+    require_license_params
+
+    create_metric(
+      @cookbook_version,
+      QualityMetric.license_metric,
+      params[:license_failure],
+      params[:license_feedback]
+    )
+
+    head 200
+  end
+
+  rescue_from ActionController::ParameterMissing do |e|
+    error(
+      error_code: t('api.error_codes.invalid_data'),
+      error_messages: [t("api.error_messages.missing_#{e.param}")]
+    )
+  end
+
+  private
+
+  def require_evaluation_params
+    params.require(:fieri_key)
+    params.require(:cookbook_name)
+    params.require(:cookbook_version)
+    params.require(:foodcritic_failure)
+  end
+
+  def require_collaborator_params
+    params.require(:cookbook_name)
+    params.require(:collaborator_failure)
+    params.require(:collaborator_feedback)
```

Comment:
I am curious as to why `:fieri_key` is not required for the `collaborator_evaluation` endpoint. I know the parameter gets used by the `check_authorization` filter before the controller action is called. I don't think anything we do _in_ the controller action uses the `:fieri_key`. Maybe we don't need to include `:fiery_key` param in these `require_thingie_params` filters?